src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4750,19 +4750,30 @@ public static void enableAutoCompaction() {
         }
     }
 
+    public void internalDisableNode() {
+        logger.info("Stopping transports and disabling auto compaction");
+        instance.stopTransports();
+        disableAutoCompaction();
+    }
+
+    @Override
+    public void disableNode() {
+        setMode(Mode.DISABLED, "Node has been disabled", true);
+        internalDisableNode();
+    }
+
+    @Override
     public void enableNode() {
```

Comment:
can you add a check to make sure that the node is already in disabled mode?

Reply:
I'll separate this out into `internalEnableNode` and `enableNode`. `enableNode` is also used by the `Directories` stuff. Alternatively, we could simply perform a check here to ensure that transient errors is not empty.

Reply:
I think checking for transient errors isn’t enough.

For example, if the node is joining or leaving or zombie node, if we “disable” then “enable” we will set it back to normal mode.  And I don’t know that it’s better to track the original status because, for example, disabling will interrupt joining.

Reply:
gotcha makes sense -- my latest push should account for this!